Added Get-PnPListItem ScriptBlock parameter to ByQuery parameter set

Rus7am opened the pull request:
## Type ##
- [x] Bug Fix
- [ ] New Feature
- [ ] Sample

## What is in this Pull Request ? ##
Added **ScriptBlock** parameter of the **Get-PnPListItem** cmd-let to the **ByQuery** parameter set

msftclas commented:
Hi __@Rus7am__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

erwinvanhunen commented:
Hi!

Thank you for your submission!

I noticed you added a parameterset designator to make the ScriptBlock parameter show up in in the ByQuery parameterset. However, this will not make the ScriptBlock automatically execute as the code will halt after returning the items if a query is specified. Please check the code and where appropriate make changes.

Rus7am commented:
Hi Erwin!

Sorry, I wasn't careful. Not only ScriptBlock would not execute in ByQuery parameter set, but PageSize parameter was not allowed in it which makes ScriptBlock pointless.

I updated GetListItem to allow both PageSize and ScriptBlock parameters in ByQuery parameter set. I also added unit tests for this case.

erwinvanhunen commented:
Thank you! Due to conflicts I manually merged this one with PR #850
